### ISSUE TITLE
ZArith's assembly file caml_z_x86_64_mingw64 is rejected due to non-executable stack statement

### ISSUE THREAD

**Issue by martin-neuhaeusser** [ocaml/Zarith#21]
Building ZArith with assembly instructions for the fast path on Cygwin64/mingw-w64 fails
due to the instructions for making the stack non-executable that have been introduced by 96a8242.

The GCC version used is `x86_64-w64-mingw32-gcc (GCC) 6.4.0`, the error message reads:
```
caml_z_x86_64_mingw64.S: Assembler messages:
caml_z_x86_64_mingw64.S:24: Error: junk at end of line, first unrecognized character is `-'
```
Reverting the commit solves the issue (but leaves the stack executable).
Just replacing `@progbits` by `%progbits` (as proposed in #14) does not solve the issue on Cygwin/mingw-w64.

Given the other bug report #14 that concerns 96a8242, the change appears to be rather non-portable.
Is there a better solution available?

**Comment by bschommer** [comment]
Replacing `@progbits` by `%progbits` should not really change anything. This change was purely due to the fact that `@` is the start of a comment in arm assembler, hence they need some different character for this kind of assembler directive. This seems to me more of a problem with the configure flags.

**Comment by martin-neuhaeusser** [comment]
As far as I can see, the configure script does not alter any of the assembly files. Only the `-host` (and technically, the `-noasm`) flags seem to influence which (or any, resp.) assembly files are used. 
From its name, I suppose that `caml_z_x86_64_mingw64.S` is the file that should be used on Cygwin64/mingw-w64. 

BTW: When removing the line introduced in 96a8242, the build succeeds.

**Comment by martin-neuhaeusser** [comment]
Using the fix proposed in #16 does not work either;
it seems that MinGW-w64 does not support the `--noexecstack` option, either.

**Comment by bschommer** [comment]
I did look in the wrong assembly file. For the other x86 targets this is guarded by a define which test whether the target binary format is ELF since this directive is only supported for ELF binary targets.

**Comment by martin-neuhaeusser** [comment]
Good point :-)
Indeed, if I add the same guard as in, e.g. `caml_z_x86_64.S`, the build succeeds.
Shall I create a PR?

**Comment by bschommer** [comment]
I already created a PR #22 which removes the stack annotation completely since this is not supported for COFF at all.

**Comment by martin-neuhaeusser** [comment]
Perfect! Thanks for the help!

**Comment by xavierleroy** [comment]
Fixed by #22.
